app/services/file_thawer.py

```
@@ -18,12 +18,38 @@
+        source: Path, prepared_destination: Path, final_destination: Path, stat_info: os.stat_result
+    ) -> None:
+        """Finalize a staged copy after checksum verification."""
+        if prepared_destination != final_destination:
+            prepared_destination.replace(final_destination)
+
+        shutil.copystat(str(source), str(final_destination))
+        os.utime(str(final_destination), ns=(stat_info.st_atime_ns, stat_info.st_mtime_ns))
+        source.unlink()
+
     @staticmethod
     def thaw_file(
```

Ruff (0.15.7)

[warning] Too many return statements (11 > 6)

(PLR0911)